### PR TITLE
Error instead of panic

### PR DESCRIPTION
**Pull request by nazar-pc**
This is needed for https://github.com/paritytech/polkadot-sdk/pull/2155

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 [package]
 name = "fdlimit"
-version = "0.2.1"
+version = "0.3.0"
```
Comment (liamaharon) [inline Cargo.toml:3]: This is a breaking change to the API, should it be a major bump?

Comment (nazar-pc) [reply Cargo.toml:3]: It isn't at `1.x` yet, so minor bump is appropriate here. Unless everyone is happy to go to 1.0 with this PR.

Comment (bkchr) [reply Cargo.toml:3]: API breakage is only a minor version bump on < 1.0.0

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@liamaharon can you quickly test if this compiles on Mac?

**Comment by liamaharon** [comment]
> @liamaharon can you quickly test if this compiles on Mac?

It compiles

**Review by liamaharon — APPROVED** [review]

**Comment by bkchr** [comment]
@nazar-pc published.

**Pull request merged**